lib/pandarus/api_base.rb

```diff
@@ -79,7 +79,12 @@ def underscored_flatten_hash(hash)
     end
 
     def dot_flatten_hash(hash)
-      Hash[dot_flatten_recur(hash)]
+      dot_flatten_hash = dot_flatten_recur(hash)
+      if dot_flatten_hash.is_a?(Array) and dot_flatten_hash.length.even?
```

Comment:
`dot_flatten_hash` - always will be an array,  useless condition `dot_flatten_hash.is_a?(Array)`